src/juju/actions.js

```diff
@@ -109,9 +110,9 @@ export function fetchModelStatus(modelUUID) {
   the supplied model info call.
   @param {String} modelInfo The response from a modelInfo call.
 */
-export function addControllerCloudRegion(modelInfo) {
+export function addControllerCloudRegion(wsControllerURL, modelInfo) {
   return async function addControllerCloudRegion(dispatch, getState) {
-    const controllers = getState()?.juju?.controllers;
+    const controllers = getState()?.juju?.controllers[wsControllerURL];
     const model = modelInfo.results[0].result;
     const updatedControllers = cloneDeep(controllers).map((controller) => {
```

Comment:
You might want to check controllers is not null now your referencing an item in the array

Reply:
Good call, done.